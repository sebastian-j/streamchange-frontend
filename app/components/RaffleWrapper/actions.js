/*
 *
 * RaffleWrapper actions
 *
 */

import {
  CHANGE_ANIMATION,
  CHANGE_DURATION,
  CHANGE_VISIBILITY,
} from './constants';

export function changeAnimation(a) {
  return {
    type: CHANGE_ANIMATION,
    animationType: a,
  };
}

export function changeAnimationDuration(t) {
  return {
    type: CHANGE_DURATION,
    animationDuration: t,
  };
}

export function changeVisibility(v) {
  return {
    type: CHANGE_VISIBILITY,
    isOpen: v,
  };
}
