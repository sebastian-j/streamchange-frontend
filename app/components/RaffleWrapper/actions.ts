/*
 *
 * RaffleWrapper actions
 *
 */
import { action } from 'typesafe-actions';
import ActionTypes from './constants';

export const changeAnimation = (animationType: number) =>
  action(ActionTypes.CHANGE_ANIMATION, animationType);

export const changeAnimationDuration = (animationDuration: number) =>
  action(ActionTypes.CHANGE_DURATION, animationDuration);

export const changeVisibility = (isOpen: boolean) =>
  action(ActionTypes.CHANGE_VISIBILITY, isOpen);
