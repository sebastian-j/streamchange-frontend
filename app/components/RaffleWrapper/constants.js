/*
 *
 * RaffleWrapper constants
 *
 */

export const CHANGE_ANIMATION = 'app/RaffleWrapper/CHANGE_ANIMATION';
export const CHANGE_VISIBILITY = 'app/RaffleWrapper/CHANGE_VISIBILITY';
