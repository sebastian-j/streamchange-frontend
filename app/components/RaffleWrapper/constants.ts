/*
 *
 * RaffleWrapper constants
 *
 */
enum ActionTypes {
  CHANGE_ANIMATION = 'app/RaffleWrapper/CHANGE_ANIMATION',
  CHANGE_DURATION = 'app/RaffleWrapper/CHANGE_DURATION',
  CHANGE_VISIBILITY = 'app/RaffleWrapper/CHANGE_VISIBILITY',
}

export default ActionTypes;
