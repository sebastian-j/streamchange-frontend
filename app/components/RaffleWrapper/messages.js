/*
 * RaffleWrapper Messages
 *
 * This contains all the text for the RaffleWrapper component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'streamchange.components.RaffleWrapper';

export default defineMessages({
  animationDuration: {
    id: `${scope}.animationDuration`,
    defaultMessage: 'Animation duration (seconds)',
  },
  noUserSelected: {
    id: `${scope}.noUserSelected`,
    defaultMessage: 'No user selected',
  },
  startBtn: {
    id: `${scope}.startBtn`,
    defaultMessage: 'Roll it!',
  },
});
