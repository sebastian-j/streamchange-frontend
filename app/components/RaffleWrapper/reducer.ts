/*
 *
 * RaffleWrapper reducer
 *
 */
import ActionTypes from './constants';
import { ContainerState, ContainerActions } from './types';

export const initialState: ContainerState = {
  animationType: 0,
  animationDuration: Number(localStorage.getItem('gv-animationDuration')) || 7,
  isOpen: false,
};

/* eslint-disable default-case, no-param-reassign */
function raffleWrapperReducer(
  state: ContainerState = initialState,
  action: ContainerActions,
): ContainerState {
  switch (action.type) {
    case ActionTypes.CHANGE_ANIMATION:
      return {
        animationType: action.payload,
        animationDuration: state.animationDuration,
        isOpen: state.isOpen,
      };
    case ActionTypes.CHANGE_DURATION:
      localStorage.setItem('gv-animationDuration', String(action.payload));
      return {
        animationType: state.animationType,
        animationDuration: action.payload,
        isOpen: state.isOpen,
      };
    case ActionTypes.CHANGE_VISIBILITY:
      return {
        animationType: state.animationType,
        animationDuration: state.animationDuration,
        isOpen: action.payload,
      };
    default:
      return state;
  }
}

export default raffleWrapperReducer;
