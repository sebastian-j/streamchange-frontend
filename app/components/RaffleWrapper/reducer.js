/*
 *
 * RaffleWrapper reducer
 *
 */
import produce from 'immer';

import { CHANGE_ANIMATION, CHANGE_VISIBILITY } from './constants';

export const initialState = {
  animationType: 0,
  isOpen: false,
};

/* eslint-disable default-case, no-param-reassign */
const raffleWrapperReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_ANIMATION:
        draft.animationType = action.animationType;
        break;
      case CHANGE_VISIBILITY:
        draft.isOpen = action.isOpen;
        break;
    }
  });

export default raffleWrapperReducer;
