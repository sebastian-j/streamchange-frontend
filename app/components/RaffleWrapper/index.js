import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { FormattedMessage } from 'react-intl';
import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';

import messages from './messages';
import CSGORaffle from '../CSGORaffle';
import FortuneWheelRaffle from '../FortuneWheelRaffle';
import NumericInput from '../NumericInput';
import db from '../YoutubeWorker/db';

const StartButton = styled.button`
  background-color: ${props => props.theme.buttonBackground};
  border: 1px solid ${props => props.theme.color};
  color: ${props => props.theme.buttonTextColor};
  cursor: pointer;
  margin-top: 10px;
  padding: 8px 12px;
  width: 100%;
  &:hover {
    background-color: ${props => props.theme.buttonBackgroundHover};
    color: ${props => props.theme.buttonTextColorHover};
  }
`;

const StyledFormControl = styled(FormControl)`
  width: 100%;
  input {
    color: ${props => props.theme.staticTextColor};
  }
  label {
    color: ${props => props.theme.inputLabel};
  }
  label.Mui-focused {
    color: ${props => props.theme.inputLabelFocused};
  }
  span,
  svg {
    color: ${props => props.theme.staticTextColor};
  }
`;
const RaffleWrapper = props => {
  const [isOpen, setIsOpen] = useState(false);
  const [noUsers, setNoUsers] = useState(false);
  const [duration, setDuration] = useState(
    Number(localStorage.getItem('gv-animationDuration')) || 7,
  );
  const [raffleType, setRaffleType] = useState(0);

  const openDialog = () => {
    db.table('users')
      .filter(user => user.isEligible === true)
      .toArray()
      .then(items => {
        if (items.length > 0) {
          setIsOpen(true);
        } else {
          setNoUsers(true);
          setTimeout(() => setNoUsers(false), 3000);
        }
      });
  };

  const closeDialog = () => {
    setIsOpen(false);
  };

  const changeDuration = value => {
    setDuration(value);
    localStorage.setItem('gv-animationDuration', String(value));
  };

  return (
    <div>
      <StyledFormControl margin="normal">
        <InputLabel id="animation-select">
          <FormattedMessage {...messages.raffleType} />
        </InputLabel>
        <Select
          onChange={event => setRaffleType(event.target.value)}
          value={raffleType}
        >
          <MenuItem value={0}>
            <FormattedMessage {...messages.raffleTypeCS} />
          </MenuItem>
          <MenuItem value={1}>
            <FormattedMessage {...messages.raffleTypeWheel} />
          </MenuItem>
        </Select>
      </StyledFormControl>
      <FormattedMessage {...messages.animationDuration}>
        {label => (
          <NumericInput
            label={label}
            minValue={1}
            maxValue={600}
            value={duration}
            onChange={ret => changeDuration(ret)}
          />
        )}
      </FormattedMessage>
      <StartButton disabled={noUsers} type="button" onClick={openDialog}>
        {noUsers ? (
          <FormattedMessage {...messages.noUserSelected} />
        ) : (
          <FormattedMessage {...messages.startBtn} />
        )}
      </StartButton>
      {isOpen && raffleType === 0 && (
        <CSGORaffle
          duration={duration}
          onClose={closeDialog}
          onWin={props.onWin}
        />
      )}
      {isOpen && raffleType === 1 && (
        <FortuneWheelRaffle
          duration={duration}
          onClose={closeDialog}
          onWin={props.onWin}
        />
      )}
    </div>
  );
};

RaffleWrapper.propTypes = {
  onWin: PropTypes.func.isRequired,
};

export default RaffleWrapper;
