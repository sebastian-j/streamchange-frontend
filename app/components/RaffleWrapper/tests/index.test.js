import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import RaffleWrapper from '../index';

const renderer = new ShallowRenderer();

describe('<RaffleWrapper />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<RaffleWrapper onWin={() => 0} />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
