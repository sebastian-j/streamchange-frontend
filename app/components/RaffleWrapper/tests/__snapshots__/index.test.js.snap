// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RaffleWrapper /> should render and match the snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root RaffleWrapper__StyledFormControl-emso03-1 hJudSC MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink={true}
      id="animation-select"
    >
      <span>
        Animation
      </span>
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <div
        aria-haspopup="listbox"
        className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span>
          CS:GO case opening
        </span>
      </div>
      <input
        aria-hidden={true}
        className="MuiSelect-nativeInput"
        onAnimationStart={[Function]}
        onChange={[Function]}
        required={false}
        tabIndex={-1}
        value={0}
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
  <div
    className="NumericInput__Container-sc-2ueyqw-0 eNCQqt"
  >
    <div
      className="NumericInput__Label-sc-2ueyqw-1 dNgzEf"
    >
      Animation duration (seconds)
    </div>
    <button
      className="NumericInput__Button-sc-2ueyqw-2 egZSpV"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <input
      className="NumericInput__NumberDisplay-sc-2ueyqw-3 hBwKoe"
      onBlur={[Function]}
      onChange={[Function]}
      value={7}
    />
    <button
      className="NumericInput__Button-sc-2ueyqw-2 ioDaXV"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <button
    className="RaffleWrapper__StartButton-emso03-0 ffgcbq"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <span>
      Roll it!
    </span>
  </button>
</div>
`;
